go.mod

```diff
@@ -164,28 +167,30 @@ require (
 	go.uber.org/atomic v1.7.0 // indirect
 	go.uber.org/multierr v1.6.0 // indirect
 	golang.org/x/oauth2 v0.0.0-20220608161450-d0670ef3b1eb // indirect
-	golang.org/x/sys v0.3.0 // indirect
-	golang.org/x/term v0.3.0 // indirect
-	golang.org/x/text v0.5.0 // indirect
+	golang.org/x/sys v0.5.0 // indirect
+	golang.org/x/term v0.5.0 // indirect
+	golang.org/x/text v0.7.0 // indirect
 	gomodules.xyz/jsonpatch/v2 v2.2.0 // indirect
 	google.golang.org/appengine v1.6.7 // indirect
 	google.golang.org/protobuf v1.28.1 // indirect
 	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c // indirect
 	gopkg.in/go-playground/validator.v9 v9.31.0 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
-	k8s.io/klog/v2 v2.80.1 // indirect
-	k8s.io/kube-openapi v0.0.0-20220803162953-67bda5d908f1 // indirect
+	k8s.io/klog/v2 v2.90.0 // indirect
+	k8s.io/kube-openapi v0.0.0-20221012153701-172d655c2280 // indirect
 	kubevirt.io/controller-lifecycle-operator-sdk v0.2.1 // indirect
-	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
+	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/kustomize/api v0.12.1 // indirect
 	sigs.k8s.io/kustomize/kyaml v0.13.9 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
 
 replace (
+	k8s.io/apimachinery => k8s.io/apimachinery v0.25.2
```

Comment:
why this replace?

Reply:
`apimachinery` update is required since need to update `github.com/openshift/api` to include the mandatory API changes added for this issue.
Updating `apimachinery to v0.26.1` adds a new arg to this func `ValidateLabelSelector` in `k8s.io/apimachinery/pkg/apis/meta/v1/validation` package  and it causes below failure.

```# sigs.k8s.io/cluster-api/api/v1beta1
vendor/sigs.k8s.io/cluster-api/api/v1beta1/machineset_types.go:184:83: not enough arguments in call to metav1validation.ValidateLabelSelector
        have (*"k8s.io/apimachinery/pkg/apis/meta/v1".LabelSelector, *field.Path)
        want (*"k8s.io/apimachinery/pkg/apis/meta/v1".LabelSelector, "k8s.io/apimachinery/pkg/apis/meta/v1/validation".LabelSelectorValidationOptions, *field.Path)
make: *** [hypershift] Error 2
```

If I update `cluster-api` to latest version which updates `controller-runtime` which causes issue in `cluster-api-provider-aws` which still uses lower version of `controller-runtime`.

So by adding this change only I am able to avoid the issues and successfully build the code.

Reply:
can we move all to .26 / latest pointing capa to main branch?